samri/pipelines/utils.py

```diff
@@ -72,15 +72,25 @@ def datasource_exclude(in_files, excludes, output="files"):
 	else:
 		exclude_criteria=[]
 		for key in excludes:
+			chars=" -_"
 			if key in "subjects":
 				for i in excludes[key]:
-					exclude_criteria.append("sub-"+str(i))
+					if i in chars:
```

Comment:
Hi, and sorry for the delay.

I really don't understand how this is intended to work. Basically the excludes variable is a dictionary (described in the docstring) which is used to exclude scans from the analysis. What you seem to be doing here is trying to not exclude based on exclusion criteria which contain the BIDS no-go characters. This is not really desirable. What we would rather be looking to do is convert (or delete) no-go characters for the scans which are analyzed --- and this has nothing to do with the excludes criteria.

Additionally, the expression `if i in chars` will never evaluate to true, since it would require a scan/subject/session identifier to be a subset of the `"" -_"` string - which it is very unlikely to ever be. Am I misunderstanding anything?